DEVELOPER.md

```diff
@@ -73,93 +85,315 @@ In JSON-Config you can use the sendTo component to send a message to your adapte
-The message contains the Query-parameters provided in the request as object.
+As soon as the authentication is done, the user is redirected to the given Redirect URL the magic admin route `/oauth2_callbacks/<adapter namespace>/` is opened in the browser. This will send a message with the defined name `oauth2Callback` to the provided adapter instance.
+The message contains the Query-parameters provided in the request as an object.
 
-Parameters like "?state=ABC&code=DEF" would be received in the message as
+Parameters like `?state=ABC&code=DEF` would be received in the message as
 
 ```json
 {
-    "state":"ABC",
-    "code":"DEF"
+    "state": "ABC",
+    "code": "DEF"
 }
 ```
 
-With these information you can easily exchange the code for an access token and so on and complete the authentication process.
+With this information, you can easily exchange the code for an access token and so on and complete the authentication process.
 
-You need to return a response to the "oauth2Callback" message to allow Admin to show a proper page to the user.
+You need to return a response to the `oauth2Callback` message to allow Admin to show a proper page to the user.
 
 ```javascript
 if (error) {
-    obj.callback && adapter.sendTo(obj.from, obj.command, {error: `Authentiction error: ${error}. Please try again.`}, obj.callback);
+    obj.callback &&
+        adapter.sendTo(
+            obj.from,
+            obj.command,
+            { error: `Authentiction error: ${error}. Please try again.` },
+            obj.callback,
+        );
 } else {
-    obj.callback && adapter.sendTo(obj.from, obj.command, {result: 'Tokens updated successfully. Please reload configuration.'}, obj.callback);
+    obj.callback &&
+        adapter.sendTo(
+            obj.from,
+            obj.command,
+            { result: 'Tokens updated successfully. Please reload configuration.' },
+            obj.callback,
+        );
 }
 ```
 
-## How to show help information in configuration dialog (Material-UI)
+## Dynamic Interface in Notifications
+
+Starting from [js-controller v6.x](https://github.com/ioBroker/ioBroker.js-controller/blob/master/README.md#notification-system), developers can send system notifications that are visible in the admin interface.
+
+In js-controller v7.x, this feature has been extended to allow adapters to send not only static information but also dynamic GUIs using the [JSON-Config schema](packages/jsonConfig/SCHEMA.md).
 
-Help can be a combination of tooltip, link and text.
+### Generating Notifications
 
+To display a dynamic GUI (with tables, buttons, colored texts, etc.), developers should generate notifications in the following form:
+
+```js
+await adapter.registerNotification(
+    'adapterName',
+    'notificationScope', // defined in io-package.json => notifications
+    I18n.translate('New dynamic notification: %s', new Date().toLocaleString()),
+    {
+        // contextData indicates that your adapter supports dynamic GUI
```

Comment:
I think this will make problems if later on `contextData` is used by other adapters, because then admin does not know if it is for admin or not. I think in the meeting we said we will have structure like `contextData.admin.offlineMessage`, so presence of `admin` or another property like `dynamicNotification` in `contextData` would mean dynamic notification support is given.

Reply:
or something like "contextData.supportsAdmin = tue or such?

Reply:
Best would be `contextData.admin.notifications` if this exists Adapter supports Dynamic notifications for this notification, and offlineMessage then as `contextData.admin.notifications.offlineMessage` than it would be a 100 % Future Proof also if Admin had at some time other use for `contextData’ and it would follow the regommendation to use Adapter Name as property for contextData!

Reply:
I am not the fan of extra long structures. If we want to say that it is for admin then `contextData.admin.notifications.offlineMessage` => `adminNotification.offlineMessage`

Reply:
With the deep structure it is extensible for future things. If you do it flat this is then harder to achieve.

Reply:
@Apollon77 @foxriver76 I changed the description